src/migrations/20231123155649-favor-permission-name-over-id.js

```diff
@@ -0,0 +1,253 @@
+exports.up = function (db, cb) {
+    db.runSql(
+        `
+-- STEP 1: Ensure 'permission' column contains unique values
+ALTER TABLE permissions
+ADD CONSTRAINT permission_unique UNIQUE (permission);
+
+-- STEP 2: Add a new column for the new foreign key
+ALTER TABLE role_permission
+ADD COLUMN permission text;
+
+-- STEP 3: Populate the new column with corresponding values from 'permissions'
+UPDATE role_permission rp
+SET permission = p.permission
+FROM permissions p
+WHERE rp.permission_id = p.id;
+
+-- STEP 4: Drop the 'id' primary key constraint
+ALTER TABLE permissions
+DROP CONSTRAINT permissions_pkey;
+
+-- STEP 5: Add the 'permission' primary key constraint
+ALTER TABLE permissions
+ADD PRIMARY KEY (permission);
+
+-- STEP 6: Drop the old 'permission_id' foreign key column
+ALTER TABLE role_permission
+DROP COLUMN permission_id;
```

Comment:
Don't do this in the same release, this breaks the expand and contract pattern: https://openpracticelibrary.com/practice/expand-and-contract-pattern/ 

I believe we need to wait until the next minor release to run this query.

Reply:
Good catch, addressed in https://github.com/Unleash/unleash/pull/5409/commits/14c1cd9476366430cb1da62e970113d97afe615e and moved to another PR: https://github.com/Unleash/unleash/pull/5413.